docs/src/development/variables/set-variables.md

```diff
@@ -0,0 +1,158 @@
+
+One example would be to define a Node.js application's build on a production branch (`NODE_ENV=production`),
+but use development mode (`NODE_ENV=development`) for each of your development environments.
+Assuming you have a `main` environment for production and then a `staging` environment with more child environments for development,
+run the following commands:
+
+```bash
+platform variable:create -l environment -e main --prefix env: --name NODE_ENV --value production --visible-build true --inheritable false
+platform variable:create -l environment -e staging --prefix env: --name NODE_ENV --value development --visible-build true --inheritable true
+```
+
+Now `NODE_ENV` is `production` on the default branch but `development` on `staging` and each of its child environments.
+Note that build visible environment variables change the application's build configuration ID:
+value updates trigger a rebuild of the application in the same way that a commit would.
+
+## Set variables via script
+
+You can also provide a `.environment` file as in your application root (next to the `.platform.app.yaml` file for that app).
+This file runs as a bash script when the container starts and on all SSH logins.
+It can be used to set any environment variables directly, such as the PATH variable.
+
+For example, the following `.environment` file allows any executable installed in the `vendor/bin` directory
+(such as executables installed using Composer)
+to be run regardless of the current directory:
+
+```bash {location=".environment"}
+export PATH=/app/vendor/bin:$PATH
+```
+
+Note that the file is sourced after all other environment variables above are defined and so they're available to the script.
+This also means the `.environment` script has the last word on environment variable values and can override anything it wants to.
+
+## Map variables
+
+If your app needs different names for environment variable than those set by Platform.sh, which is common for database connections,
+map the Platform.sh's variable names to those required by the application.
+Do this in the app with the help of the Config Reader libraries or via a shell script.
+
+For example, the following [`.environment` script](#set-variables-via-script) exports variables that are visible to the application.
+It uses the `jq` library, which is included in all app containers for this purpose.
+
+```bash {location=".environment"}
+export APP_DATABASE_HOST=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".database[0].host")
+export APP_DATABASE_PORT=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".database[0].port")
+export APP_DATABASE_NAME=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".database[0].path")
+export APP_DATABASE_USER=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".database[0].username")
+export APP_DATABASE_PASSWORD=$(echo $PLATFORM_RELATIONSHIPS | base64 --decode | jq -r ".database[0].password")
+```
+
```

Comment:
```suggestion
To use it:
1. Add the `.environment` file to your project and adapt the exported variables if needed.
1. [Set the environment variables](#create-environment-specific-variables) (`APP_DATABASE_HOST`, `APP_DATABASE_PORT`, ...).

Now, the values can be accessed in the `env` directly.
```

Reply:
Or is my understanding of the feature, wrong?

Reply:
The `.environment` file should set the variables as environment variables on its own. That's what it's there for, as far as I know.